src/domain/users/users.repository.spec.ts

```diff
@@ -63,4 +72,115 @@ describe('UsersRepository', () => {
+        user: mockAuthenticatedWallet.user,
+        address: addressToRemove,
+      });
+    });
+
+    it('should throw an UnauthorizedException if the auth payload is empty', async () => {
+      const authPayload = new AuthPayload();
+      const addressToRemove = getAddress(faker.finance.ethereumAddress());
+
+      await expect(
+        usersRepository.deleteWalletFromUser({
+          authPayload,
+          walletAddress: addressToRemove,
+        }),
+      ).rejects.toThrow(UnauthorizedException);
+    });
+
+    it('should throw a BadRequestException if there is only one wallet', async () => {
+      const authPayloadDto = authPayloadDtoBuilder().build();
+      const authPayload = new AuthPayload(authPayloadDto);
+      const addressToRemove = getAddress(faker.finance.ethereumAddress());
+
+      const mockAuthenticatedWallet = walletBuilder().build();
+      mockWalletRepository.findOne.mockResolvedValueOnce(
+        mockAuthenticatedWallet,
+      );
+      mockWalletRepository.count.mockResolvedValue(1);
+
+      await expect(
+        usersRepository.deleteWalletFromUser({
+          authPayload,
+          walletAddress: addressToRemove,
+        }),
+      ).rejects.toThrow(BadRequestException);
+    });
+
+    it('should throw an ConflictException if the user tries to remove the currently authenticated wallet', async () => {
+      const authPayloadDto = authPayloadDtoBuilder().build();
+      const authPayload = new AuthPayload(authPayloadDto);
+
+      await expect(
+        usersRepository.deleteWalletFromUser({
+          authPayload,
+          walletAddress: authPayload.signer_address as `0x${string}`,
+        }),
+      ).rejects.toThrow(
+        new ConflictException('Cannot remove the current wallet'),
+      );
+    });
+
+    it('should throw a NotFoundException if the user is not found', async () => {
+      const authPayloadDto = authPayloadDtoBuilder().build();
+      const authPayload = new AuthPayload(authPayloadDto);
+
+      const addressToRemove = getAddress(faker.finance.ethereumAddress());
+
+      mockWalletRepository.findOne.mockResolvedValueOnce(null);
+
+      await expect(
+        usersRepository.deleteWalletFromUser({
+          authPayload,
+          walletAddress: addressToRemove,
+        }),
+      ).rejects.toThrow(new NotFoundException('User not found'));
+    });
+
+    it('should throw a NotFoundException if the wallet does not exist for the user', async () => {
+      const authPayloadDto = authPayloadDtoBuilder().build();
+      const authPayload = new AuthPayload(authPayloadDto);
+
+      const addressToRemove = getAddress(faker.finance.ethereumAddress());
+
+      const mockAuthenticatedWallet = walletBuilder().build();
+      mockWalletRepository.findOne.mockResolvedValueOnce(
+        mockAuthenticatedWallet,
+      );
+      mockWalletRepository.delete.mockResolvedValue({ affected: 0, raw: {} });
+      mockWalletRepository.count.mockResolvedValue(2);
```

Comment:
This should successfully delete as there are >1 wallets attached to the user. This test should have a count of `1`, and a second test be added to test successful deletion should the count be >1.

Reply:
In this case we want to check that even though other wallets exist (count > 1) the wallet we are trying to delete does not so essentially testing the `deleteResult.affected` case at the very end.